### PR TITLE
feat(prgroom): status verb + §4.6 merge-gate envelope + §4.4 human-review (8.11)

### PR DESCRIPTION
**Pull request by scotthamilton77**
## Scope

Bead **8.11** — the read-only **`status` verb** + the stable **`status --json` merge-gate envelope** (§4.6) + the **§4.4 human-review derivation**. The final M1 bead and the second half of the M1 demo (`poll` then `status --json`).

**Stacked on 8.9a** (base = `worktree-prgroom-8.9a-poll`, PR #139). This PR's diff is only 8.11's work; merge **after** the stack below it.

## What's in this PR

- **`lifecycle/status.py`** — `build_status(state, human_review)` builds the §4.6 envelope (pure, never writes): `merge_gates{phase_is_quiesced, last_error_clear, no_blocker_items, human_review_satisfied}`, `human_review{required, satisfied_by, candidates_seen}`, `auto_merge_eligible` (= AND of the four gates), plus `items_summary`/`reviewers`/`ci_state`/timestamps.
- **`lifecycle/human_review.py`** (§4.4/§4.6) — `derive_human_review(labels, reviews)` (pure) + `fetch_human_review_inputs(gh, ref)` (the live gh enrichment). `satisfied_by` precedence: `'label'` > `'approval:{login}'` (first non-Bot APPROVED, API order) > `None`. Load-bearing **bot filter** (`user.type=='Bot'` OR `login` ends `[bot]`) — a `github-copilot[bot]` approval does NOT satisfy. `candidates_seen` carries one row per approval candidate + its filter outcome (operator-debuggability). `human_review_satisfied` is **derived, never persisted, never blocks quiescence**.
- **CLI `status` wiring** (`cli.py`) — default path is the **lock-free** read (§3.3 carve-out); `--locked` wraps it in `with_lock` (exit 75 under contention). `--json` emits the envelope; default emits a human-readable form.
- **`errors.py`** — new `PRECONDITION_NO_STATE` (exit 2) — see decision below.

## Two ratified design decisions (flagged for review — not bugs)
1. **`human_review` is a LIVE gh fetch** (2 read-only GETs: labels + reviews), not a pure state read. The state schema stores neither labels nor approval-actor-type (`ReviewerStatus` has no `approved` value), and `data-view.md` pins human_review as a live enrichment — a pure-state derivation is impossible. The gh calls are read-only, so the §3.3 lock-free-read safety still holds (no state write).
2. **`PRECONDITION_NO_STATE` (exit 2)** for `status` on a never-polled PR. `PRECONDITION_NO_ITEMS` is a *no-work* code → exit 0, which would falsely tell a scheduler "done." Exit 2 ("run poll first") is the only correct semantics; adding a code is non-breaking per the registry contract.

## In-house adversarial review (3-lens) — applied before this PR
Lens-1 GO (envelope exact-matches §4.6; precedence + bot-filter correct; `last_error_clear` is the §4.6 null/empty check). Fixes (commit `1d4227a`):
- **HIGH** — the human-review gh fetch now maps a 404 → terminal `PrgroomError` (exit 77), mirroring poll.py — previously a 404 (PR deleted mid-run) escaped as a raw traceback.
- **MED** — the missing-state test now asserts ZERO gh calls before the state read (proving `status` raises `PRECONDITION_NO_STATE` before any gh I/O).
- **LOW** — promoted the blocker-disposition set to one public `BLOCKER_DISPOSITIONS` (status + quiescence import it; no silent-drift); completed the `auto_merge_eligible` truth table.

## Ground-truth references
`docs/plans/2026-05-12-prgroom-cli-design.md` §4.4, §4.6, §3.3; HLD `data-view.md`.

## Intentionally out of scope
`request_human_review_if_needed` (the label POST — run-loop bead's escalation path); the future merge-gate verb; the `_run` aggregator.

## Test Plan
- [x] `make ci-prgroom` — exit 0: ruff ✓, format ✓, mypy --strict ✓ (31 files), **467 tests / 100% branch coverage**, pip-audit ✓, `prgroom --help` ✓.
- [ ] Copilot review of this PR.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: packages/prgroom/src/prgroom/lifecycle/human_review.py [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+    required = _REQUIRED_LABEL in lowered
+    label_satisfies = _APPROVED_LABEL in lowered
+
+    candidates: list[ApprovalCandidate] = []
+    first_human: str | None = None
+    for review in reviews:
+        if str(review.get("state", "")) != _APPROVED_STATE:
+            continue
+        login = str((review.get("user") or {}).get("login", ""))
+        bot = _is_bot(review)
```
[the remark this answers is not in the document]
Comment (scotthamilton77) [reply packages/prgroom/src/prgroom/lifecycle/human_review.py]: Fixed in 16d69c1. derive_human_review now classifies each APPROVED review on a reason-precedence ladder: bot > no-login > counted-human. A loginless/anonymous approval is non-counted (reason='no-login'), never sets satisfied_by, and the first VALID human still wins. Tests cover empty login, absent user object, and the loginless-then-human precedence.

File: docs/plans/2026-06-09-prgroom-status-merge-gate.md [hunk 2]
```diff
@@ -0,0 +1,843 @@
...
+
+**Architecture:** Three layers. (1) `lifecycle/human_review.py` — a **pure** derivation (`derive_human_review`) over already-fetched gh inputs (labels + review candidates) producing the `human_review` block (`required`, `satisfied_by`, `candidates_seen`) plus a thin gh-enrichment fetch that turns two REST GETs into those inputs. (2) `lifecycle/status.py` — assembles the §4.6 envelope from the in-memory `PRGroomingState` + the human-review block, derives the four `merge_gates` bools and `auto_merge_eligible`. (3) `cli.py` `status` wiring — default path is an **unlocked** `store.read`; `--locked` wraps the read in `with_lock`; renders `--json` or a human-readable default. Nothing here is persisted to state.
+
+**Tech Stack:** Python 3.14, typer, dataclasses, the existing `prsession` state/enums/`Store`, `lifecycle.quiescence` (gates), `lifecycle.locking.with_lock`, `gh.GhClient`, `errors` registry.
```
[the remark this answers is not in the document]
Comment (scotthamilton77) [reply docs/plans/2026-06-09-prgroom-status-merge-gate.md]: Fixed in 16d69c1. Plan now reads 'Python >=3.11 (the package's declared requires-python; developed on 3.14)', matching pyproject.toml.

File: docs/plans/2026-06-09-prgroom-status-merge-gate.md [hunk 3]
```diff
@@ -0,0 +1,843 @@
...
+- **Bot filter signal:** `user.type == "Bot"` (the §4.4 pinned signal "reviewer.actor.type != Bot"). GitHub's reviews API returns `user.type: "Bot"` for app/bot reviewers (`github-copilot[bot]`). A `login` ending in `[bot]` is a defensive secondary signal (covers payloads where `type` is absent).
+- **`satisfied_by` precedence:** `"label"` (a `human-approved` label, case-insensitive) > `"approval:{login}"` (FIRST non-bot APPROVED review, in API order) > `None`.
+- **`candidates_seen`:** one row per APPROVED review candidate — `{"login", "approved": true, "counted": bool, "reason": str}`. `counted` is true for the non-bot approval(s); `reason` is `"bot"` for a filtered bot, `""` for a counted human. Label-satisfaction does not manufacture a candidate row (candidates are PR-approval reviews only, per §4.6).
+- **`status` against a never-polled PR:** the lock-free read raises `StateNotFoundError`. Re-raised as `PreconditionError(PRECONDITION_NO_ITEMS)` (its registry `how` = "run `poll` first" is the exact remediation) with a status-specific detail. Reuses an existing code rather than expanding the §3.7 registry (out of scope).
```
[the remark this answers is not in the document]
Comment (scotthamilton77) [reply docs/plans/2026-06-09-prgroom-status-merge-gate.md]: Fixed in 16d69c1. Rewrote the bullet to the shipped PRECONDITION_NO_STATE (exit 2, 'run poll first'), documenting why reusing PRECONDITION_NO_ITEMS was rejected (it's a no-work code -> exit 0, which would falsely signal 'done'). Also swept the doc's illustrative snippets so the plan is self-consistent with the shipped code.

File: packages/prgroom/src/prgroom/cli.py [hunk 4]
```diff
@@ -186,9 +189,51 @@ def wait(pr: str = typer.Argument(..., help="PR ref.")) -> None:
...
+                raise PreconditionError(
+                    ErrorCode.PRECONDITION_NO_STATE,
+                    detail=ref.display(),
+                ) from exc
+
+        state = with_lock(store, ref, _read) if locked else _read()
+        labels, reviews = fetch_human_review_inputs(gh, ref)
+        human_review = derive_human_review(labels=labels, reviews=reviews)
+        envelope = build_status(state, human_review)
+        _render_status(envelope, json_out=json_out)
```
[the remark this answers is not in the document]
Comment (scotthamilton77) [reply packages/prgroom/src/prgroom/cli.py:237]: Fixed in 9b951d6. Moved gh construction to after the state read succeeds (inside the lock scope, before fetch_human_review_inputs). Both fast-fail paths now precede gh setup: NO_STATE (raised in the read) and LOCK_HELD (raised on the pre-read acquire). Tightened both fast-fail tests with a _build_gh spy asserting build_calls==0 (RED before the reorder, GREEN after).

File: packages/prgroom/src/prgroom/lifecycle/human_review.py [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+
+
+@dataclass(frozen=True, slots=True)
+class ApprovalCandidate:
+    """One examined PR-approval review with its bot-filter outcome (§4.6).
+
+    ``counted`` is whether this approval could satisfy the constraint (a non-bot
+    APPROVED); ``reason`` is ``"bot"`` for a filtered bot approval, ``""`` for a
+    counted human one. The row exists purely for operator debuggability — "why
+    didn't approval X count?".
```
[the remark this answers is not in the document]
Comment (scotthamilton77) [reply packages/prgroom/src/prgroom/lifecycle/human_review.py]: Fixed in af20e12. Updated the ApprovalCandidate docstring to enumerate all three reason values: 'bot' (filtered bot approval), 'no-login' (anonymous/loginless approval), and '' (counted human) — matching the round-1 loginless fix.
